### PR TITLE
save/restore workspace

### PR DESCRIPTION
**Pull request by rsbondi**
I changed button to icon to have room, but maybe a toolbar needs to be added to the design, but this is a starting point for the conversation.

### CODE REVIEW

**Comment by Bashlund** [comment]
Thanks a lot @rsbondi. I will look at this as soon as I can.

**Comment by javier-tarazaga** [comment]
@rsbondi thanks again! And fair point about the toolbar/design. I started a thread #14 to actually discuss and talk about it a new UX for the tool. Will focus on the UX in the news few days. Please feel free to add anything or suggest any new features or ideas! :)

**Comment by Bashlund** [comment]
@rsbondi I tried this out finally and it really works great! Thanks a lot for such a nice contribution.
What do you say about adding some type of warning about that the user is about to overwrite the whole workspace when uploading it?

File: src/components/projecteditor/control.js [hunk 1]
```diff
@@ -485,15 +487,66 @@ export default class DevkitProjectEditorControl extends Component {
...
                 <div class={style.notefiles}>
-                    Note: all files are stored in the browser only
+                    Note: all files are stored in the browser only, download to backup.
                 </div>
             </div>
         );
     };
 
+    _downloadWorkspace = e => {
+        this.backend.downloadWorkspace(e);
```
Comment (Bashlund) [inline src/components/projecteditor/control.js]: Missing `e.preventDefault();`

File: src/components/projecteditor/control.js [hunk 2]
```diff
@@ -485,15 +487,66 @@ export default class DevkitProjectEditorControl extends Component {
...
+    }
+
+    _clickWorkspace = (e) => {
+        document.querySelector('#wsFileInput').dispatchEvent(new MouseEvent('click')); // ref does not work fhttps://github.com/developit/preact/issues/477
```
Comment (Bashlund) [inline src/components/projecteditor/control.js]: Missing `e.preventDefault();`

File: src/components/projecteditor/control.js [hunk 3]
```diff
@@ -485,15 +487,66 @@ export default class DevkitProjectEditorControl extends Component {
...
+    }
+    _uploadWorkspace = (e) => {
+        var files = document.querySelector('#wsFileInput').files;
```
Comment (Bashlund) [inline src/components/projecteditor/control.js]: Missing `e.preventDefault();`

File: src/components/projecteditor/backend.js [hunk 4]
```diff
@@ -272,4 +272,44 @@ export default class Backend {
         localStorage.setItem("settings1.0", JSON.stringify(settings));
         if(cb) setTimeout(()=>cb(),1);
     };
+
+    downloadWorkspace = e => {
```
Comment (Bashlund) [inline src/components/projecteditor/backend.js]: Is `e` used? Could be removed?

**Pull request merged**